battery-status/battery-interface-idlharness.html

```diff
@@ -37,15 +40,22 @@
 </script>
 <script>
 "use strict";
+var t = async_test();
 var idl_array = new IdlArray();
 var idls;
+var manager;
 [].forEach.call(document.querySelectorAll('script[type=text\\/plain]'), function(node) {
-  // replace 'EventHandler' and 'unrestricted double' unrecognized by idlharness.js
-  idls = node.textContent.replace(/EventHandler/g, 'Function?').replace(/unrestricted double/g, 'double');
+  idls = node.textContent;
   idl_array[(node.className === 'untested') ? 'add_untested_idls' : 'add_idls'](idls);
 });
-idl_array.add_objects({Navigator: ['navigator'], BatteryManager: ['navigator.getBattery()']});
-idl_array.test();
+t.step(function() {
+navigator.getBattery().then(function(bm) {
```

Comment:
This assumes that 'navigator.getBattery()' is supported. If it is not supported, the test may return a timeout or behave like hang. Better add an assertion before this line?

Please also make proper indent.